ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessorTest.java

```diff
@@ -92,6 +137,95 @@ public void shouldCallTransactionValidatorWithExpectedTransactionValidationParam
+    Transaction messageToContract =
+        new TransactionTestFixture()
+            .to(Optional.of(contractAddr))
+            .sender(sending)
+            .type(TransactionType.EIP1559)
+            .maxFeePerGas(Optional.of(Wei.ONE))
+            .gasLimit(300000L)
+            .maxPriorityFeePerGas(Optional.of(Wei.ONE))
+            .createTransaction(senderKeys);
+
+    worldState.getOrCreateSenderAccount(sending).getMutable().setBalance(Wei.fromEth(1000L));
+    worldState.createAccount(contractAddr).getMutable().setCode(Bytes.fromHexString(manyJumps));
+
+    transactionProcessor.processTransaction(
+        blockchain,
+        worldState,
+        blockHeader,
+        messageToContract,
+        Address.fromHexString("1"),
+        blockHashLookup,
+        false,
+        ImmutableTransactionValidationParams.builder().build());
+
+    Account contractAccount = worldState.get(contractAddr);
+    Mockito.verify(worldState, times(1)).getContract(contractAccount);
+    Mockito.verify(loader, times(1)).load(contractAccount);
+
+    transactionProcessor.processTransaction(
+        blockchain,
+        worldState,
+        blockHeader,
+        messageToContract,
+        Address.fromHexString("1"),
+        blockHashLookup,
+        false,
+        ImmutableTransactionValidationParams.builder().build());
+
+    Mockito.verify(worldState, times(2)).getContract(contractAccount);
+    Mockito.verify(loader, times(1)).load(contractAccount);
+  }
+
+  @Test
+  public void shouldEvictOnAccountDestruct() {
+    KeyPair senderKeys = SignatureAlgorithmFactory.getInstance().generateKeyPair();
+
+    Address sending = Address.extract(senderKeys.getPublicKey());
+    Address contractAddr = Address.fromHexString("B0B0FACE");
+
+    Transaction messageToContract =
+        new TransactionTestFixture()
+            .to(Optional.of(contractAddr))
+            .sender(sending)
+            .gasLimit(30000)
+            .createTransaction(senderKeys);
+    ;
+
+    this.worldState.getOrCreateSenderAccount(sending).getMutable().setBalance(Wei.fromEth(1000L));
+    this.worldState
+        .createAccount(contractAddr)
+        .getMutable()
+        .setCode(Bytes.fromHexString(manyJumps));
+    this.worldState.commit();
+
+    transactionProcessor.processTransaction(
+        blockchain,
+        this.worldState,
+        blockHeader,
+        messageToContract,
+        Address.fromHexString("1"),
+        blockHashLookup,
+        false,
+        ImmutableTransactionValidationParams.builder().build());
+
+    Account contractAccount = this.worldState.get(contractAddr);
+    Mockito.verify(this.worldState, times(1)).getContract(contractAccount);
+    Mockito.verify(loader, times(1)).load(contractAccount);
+
+    this.worldState.deleteAccount(contractAccount.getAddress());
+    this.worldState.commit();
+    assertThat(this.worldState.get(contractAccount.getAddress())).isNull();
+    assertThat(this.cache.size()).isEqualTo(0);
+  }
```

Comment:
Is there a test case to cover CREATE2?

Reply:
[Yes there is](https://github.com/hyperledger/besu/blob/367b6a74f1da5a88c9a6b9c84565f0f6f6ffd406/ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/operations/Create2OperationTest.java#L47), but since jumpdest analysis is done on execution, not creation, it was not impacted.